src/journal-remote/journal-upload.c

```diff
@@ -448,6 +491,55 @@ static void destroy_uploader(Uploader *u) {
         sd_event_unref(u->event);
 }
 
+#if LIBCURL_VERSION_NUM >= 0x075300
+static int update_content_encoding(Uploader *u, char *acceptEncoding) {
+        bool found = false;
+        Compression algorithm;
+        CURLcode code;
+
+        assert(u);
+
+        while (acceptEncoding && !found) {
+                _cleanup_free_ char *encodingValue = NULL;
+                char *algorithmStr = NULL;
+
+                if (extract_first_word((const char **)&acceptEncoding, &encodingValue, ",", 0) < 0)
+                        return log_error_errno(SYNTHETIC_ERRNO(EINVAL), "Failed to extract Accept-Encoding header value");
+
+                if (extract_first_word((const char **)&encodingValue, &algorithmStr, ";", 0) < 0)
+                        return log_error_errno(SYNTHETIC_ERRNO(ENOMEM), "Failed to extract compression algorithm from Accept-Encoding header");
+
+                algorithm = compression_lowercase_from_string(algorithmStr);
+
+                for (size_t i = 0; i < arg_compression.size; i++) {
+                        CompressionOpts c = arg_compression.opts[i];
+                        if (c.algorithm != algorithm)
+                                continue;
+                        _cleanup_free_ char *header = NULL;
+                        struct curl_slist *headers = NULL;
+
+                        u->compression = c;
+                        header = strjoin("Content-Encoding: ", compression_lowercase_to_string(u->compression.algorithm));
+                        if (!header)
+                                return log_oom();
+
+                        headers = curl_slist_append(NULL, header);
+                        if (!headers)
+                                return log_oom();
+
+                        headers->next = u->header->next;
+                        u->header->next = NULL;
+                        curl_slist_free_all(u->header->next);
+                        u->header = headers;
```

Comment:
What do you want to do in this four lines?? Please add a short comment.

Reply:
in curl headers is a list and content type is a head of this list, this code changes list header, if new content type was received

Reply:
I asked because this code block seems to leak an existing header in `Uploader`.
Why not to append the header something like we do in `start_upload()` ?
```c
struct curl_slist *l = curl_slist_append(u->header, header);
if (!l)
        return log_oom();
u->header = l;
```
Anyway, please add a short comment in the code.